src/operator/nn/mkldnn/mkldnn_base.cc

```diff
@@ -128,12 +201,18 @@ void CommitOutput(const NDArray &arr, const mkldnn_output_t &res) {
   if (res.first == CopyBack) {
     const_cast<NDArray &>(arr).CopyFrom(*res.second);
   } else if (res.first == AddBack) {
+    auto res_memory = res.second;
+    auto target_pd = arr.GetMKLDNNData()->get_primitive_desc();
     auto mem = arr.GetMKLDNNData(res.second->get_primitive_desc());
-    CHECK(mem != nullptr);
+    if (mem == nullptr) {
+      auto tmp_memory = TmpMemMgr::Get()->Alloc(target_pd);
+      CopyMKLDNNMem(*res_memory, tmp_memory);
```

Comment:
I think you can just call GetMKLDNNDataReorder.
@pengzhao-intel tries to fix this bug, but doesn't have a test for his fix.
https://github.com/apache/incubator-mxnet/pull/11095

Reply:
the purpose of this block is coerce the input memory (`res`) to the same size as the output ndarray's memory. the issue here is that we do not have the ndarray of the input, only the memory in `res.second` so we don't have the `GetMKLDNNDataReorder` member function.

Reply:
I see. When there is a format mismatch between `arr` and `res`, you use the format of `arr`.  In this case, you may remove line 214 and 216 and have the sum written to `mem` directly. WriteInplace works in MKLDNN as long as the inputs and outputs have the same shape and format.